src/main/java/com/zendesk/maxwell/schema/SchemaStore.java

```diff
@@ -297,9 +309,23 @@ public void destroy() throws SQLException {
 		}
 	}
 
+	public boolean schemaExists(long schema_id) throws SQLException {
+		if ( this.schema_id == null )
+			return false;
+
+		ResultSet rs = connection.createStatement().executeQuery("select id from maxwell.schemas where id = " + schema_id);
+		return rs.next();
+	}
+
 	public BinlogPosition getBinlogPosition() {
 		return this.position;
 	}
 
+	private void deleteOldSchemas(Long currentSchemaId) throws SQLException {
+		Long toDelete = currentSchemaId - SCHEMAS_TO_KEEP;
```

Comment:
is this a valid assumption? (That ids are incrementing properly). Also why are we not deleting everything older?